api/src/main/java/org/apache/iceberg/catalog/Catalog.java

```diff
@@ -303,4 +303,83 @@ default boolean dropTable(TableIdentifier identifier) {
    * @throws NoSuchTableException if the table does not exist
    */
   Table loadTable(TableIdentifier identifier);
+
+  /**
+   * Instantiate a builder to either create a table or start a create/replace transaction.
+   *
+   * @param identifier a table identifier
+   * @param schema a schema
+   * @return the builder to create a table or start a create/replace transaction
+   */
+  default TableBuilder buildTable(TableIdentifier identifier, Schema schema) {
+    throw new UnsupportedOperationException(this.getClass().getName() + " does not implement buildTable");
```

Comment:
I am throwing `UnsupportedOperationException` to avoid breaking custom catalogs.

Reply:
Sounds good.